src/main/kotlin/com/bitclave/node/services/v1/OfferService.kt

```diff
@@ -115,6 +115,50 @@ class OfferService(
         })
     }
 
+    fun putBulkAdvanced(
+        owner: String,
+        offers: Array<Offer>,
+        strategy: RepositoryStrategyType
+    ): CompletableFuture<List<Long>> {
+        return supplyAsyncEx(Supplier {
+            val updatedOfferIds = offers.filter { it.id != 0L }.map { it.id }
+
+            val existedOffers = offerRepository.changeStrategy(strategy).findByIds(updatedOfferIds)
+
+            var index = 0
+            val readyForSaveOffers = offers.map {
+                val createdAt = if (it.id != 0L) existedOffers[index++].createdAt else Date()
+                Offer(
+                    it.id,
+                    owner,
+                    it.offerPrices,
+                    it.description,
+                    it.title,
+                    it.imageUrl,
+                    it.worth,
+                    it.tags,
+                    it.compare,
+                    it.rules,
+                    createdAt
+                )
+            }
+            val result = offerRepository.changeStrategy(strategy).saveAll(readyForSaveOffers)
+
+            val prices = result.mapIndexed { index, offer ->
```

Comment:
for clear code, please change name "index". because it shadowed name.

Reply:
fixed